File: csharp/src/Blockchain.Abstractions/Models/HTLCCommitTransactionPrepareRequest.cs
﻿namespace Train.Solver.Blockchain.Abstractions.Models;

public class HTLCCommitTransactionPrepareRequest
{
    public string Receiever { get; set; } = null!;

    public string[] HopChains { get; set; }

    public string[] HopAssets { get; set; }

    public string[] HopAddresses { get; set; }

    public string DestinationChain { get; set; } = null!;

    public string DestinationAsset { get; set; } = null!;

    public string DestinationAddress { get; set; } = null!;

    public string SourceAsset { get; set; } = null!;

    public long Timelock { get; set; }

    public decimal Amount { get; set; }
}

File: js/src/Blockchain/Blockchain.Abstraction/Models/TransactionBuilderModels/HTLCCommitTransactionPrepareRequest.ts
export interface HTLCCommitTransactionPrepareRequest {
  receiver: string;
  hopChains: string[];
  hopAssets: string[];
  hopAddresses: string[];
  destinationChain: string;
  destinationAsset: string;
  destinationAddress: string;
  sourceAsset: string;
  timelock: number;
  amount: number;
  id: string;
}